server/src-lib/Hasura/GraphQL/Resolve/Select.hs

```diff
@@ -44,6 +44,15 @@ withSelSet selSet f =
     res <- f fld
     return (G.unName $ G.unAlias $ _fAlias fld, res)
 
+paramsToAnnArgs :: ArgsMap -> RS.AnnArgs
```

Comment:
We've to resolve `ArgsMap` to `ColOp` here. 

```haskell
argsToColOp 
  :: ArgsMap -> Maybe RS.ColOp
```
`lookup` for `path` field in `ArgsMap` if present then resolve `AnnGValue` to an array `SQLExp` else `Nothing`. I would suggest to split path value `(path: "<path-value>")` on `.` (`T.splitOn "."`).  Convert `Text` array to `SQLExp` array expression using `SEArray`.

Reply:
So we only support simple JSON Path such as ```user.address.street```. I intend to support another JSON query spec such as array ```users[0].address```. Does it make sense? Or I just skip it.

Reply:
Nice idea! You should somehow resolve `Text` value to appropriate SQL array using `SEArray`.

Reply:
For example, `users[0].address` to `['users', 0, 'address']`